### ISSUE TITLE
"Create Octopus Release" started failing with Unrecognized command arguments: 

### ISSUE THREAD

**Issue by danielab** [OctopusDeploy/OctoTFS#224]
After Friday 11.03.22 our "Create Octopus Release" task v3 started failing in our Azure Devops pipeline. There has been a change in the Octo.exe create-release command. **An empty argument "" has appeared**. We have not done any changes since last successful run.

Previous working output from our pipeline logs of the azure devops task: 

`c:\choco\bin\Octo.exe create-release "--project=Foo.Bar API" --releaseNumber=2022.3.4.20 --channel=Default --server=http://foo-bar.biz/ --apiKey=*** --enableServiceMessages --releaseNotesFile=D:\w\6\2277\s\release-notes-4706a450-9bbb-11ec-a119-fb175bb88688.md --packageversion 2022.3.4.20`

Now, the command has changed to (see the double empty quotes after releaseNumber and before channel):

`c:\choco\bin\Octo.exe create-release "--project=Foo.Bar API" --releaseNumber=2022.3.14.20 "" --channel=Default --server=http://foo-bar.biz/ --apiKey=*** --enableServiceMessages --releaseNotesFile=D:\w\5\2500\s\release-notes-60532ff7-9080-4559-9d56-de09af4b8d2e.md --packageversion 2022.3.14.20`

```
Unrecognized command arguments: 

This error is most likely occurring while executing octo as part of an automated build process. The following doc is recommended to get some tips on how to troubleshoot this: https://g.octopushq.com/OctoexeTroubleshooting
Exit code: -1
##[error]"Failed to create release. The process 'c:\choco\bin\Octo.exe' failed with exit code 4294967295
Error: The process 'c:\choco\bin\Octo.exe' failed with exit code 4294967295
    at ExecState._setResult (D:\w\5\_tasks\OctopusCreateRelease_4e131b60-5532-4362-95b6-7c67d9841b4f\3.1.27\node_modules\azure-pipelines-task-lib\toolrunner.js:942:25)
    at ExecState.CheckComplete (D:\w\5\_tasks\OctopusCreateRelease_4e131b60-5532-4362-95b6-7c67d9841b4f\3.1.27\node_modules\azure-pipelines-task-lib\toolrunner.js:925:18)
    at ChildProcess.<anonymous> (D:\w\5\_tasks\OctopusCreateRelease_4e131b60-5532-4362-95b6-7c67d9841b4f\3.1.27\node_modules\azure-pipelines-task-lib\toolrunner.js:838:19)
    at ChildProcess.emit (events.js:198:13)
    at maybeClose (internal/child_process.js:982:16)
```

YAML definition of the Azure DevOps task:

```yaml
steps:
- task: octopusdeploy.octopus-deploy-build-release-tasks.octopus-create-release.OctopusCreateRelease@3
  displayName: 'Create Octopus Release for master: "$(build.ApplicationName)"'
  inputs:
    OctoConnectedServiceName: 'FooBar Server'
    ProjectName: '"$(build.ApplicationName)"'
    ReleaseNumber: '"$(build.Number)"'
    Channel: Default
    ChangesetCommentReleaseNotes: true
    WorkItemReleaseNotes: true
    DeploymentProgress: true
    AdditionalArguments: '--packageversion "$(build.Number)" --releasenotes "Built from commit $(build.sourceVersion)"'
  condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
```
 
We are using: Octopus Deploy Command Line Tool, version 7.4.3116
When running this command directly on local machine it works with powershell but not with unix shell.
I have tried to use the task version v4 in azure devops but with same error as above.

Any feedback is appreciated.

**Comment by johnsimons** [comment]
Hi @danielab,

There were a few changes pushed around that time, I am sorry that these are causing issues to you.
I am trying to reproduce this problem, would you be able to "Enable System Diagnostics" and send us the full raw logs?
If the raw logs have sensitive information, [email them to our support email address](https://octopus.com/support). 
<img width="236" alt="image" src="https://user-images.githubusercontent.com/122651/158701638-4ff29b31-5a5b-46bb-8aad-c01fde4db192.png">
<img width="716" alt="image" src="https://user-images.githubusercontent.com/122651/158700751-212f490e-9ac2-46c9-a0f6-4c35fca1b239.png">

I used a hosted agent and copy-pasted your yaml onto a new pipeline, and it seems to work, no extra `""`.

Just wondering what happens if you add the Installer task beforehand?
Here is what seems to work for me:
```yaml
steps:
- task: OctoInstaller@4
  inputs:
    version: '8.0.0'
- task: OctopusCreateRelease@4
  displayName: 'Create Octopus Release for master: "$(build.ApplicationName)"'
  inputs:
    OctoConnectedServiceName: 'octopus main'
    ProjectName: '"$(build.ApplicationName)"'
    ReleaseNumber: '"$(build.Number)"'
    Space: Default
    Channel: Default
    ChangesetCommentReleaseNotes: true
    WorkItemReleaseNotes: true
    DeploymentProgress: true
    AdditionalArguments: '--packageversion "$(build.Number)" --releasenotes "Built from commit $(build.sourceVersion)'
```

**Comment by danielab** [comment]
Thank you for your response!
I have sent raw logs to the support email with the subject: 
"https://github.com/OctopusDeploy/OctoTFS/issues/224 raw logs attachments"

I also tried to add the OctoInstaller with version 8.0.0 and still got the same error.
We are running on a private on premise build agent which I forgot to mention earlier.

**Comment by Clare-Octopus** [comment]
https://octopus.zendesk.com/agent/tickets/85593 - Daniels ticket with his logs included (Internal Link)

**Comment by johnsimons** [comment]
@danielab I'm still investigating this issue, thank you for the logs.

Are you able to try to remove the double quotes and run it again:
```yaml
ProjectName: '$(build.ApplicationName)'
ReleaseNumber: '$(build.Number)'
```

From your raw logs I can see the double up in quotes:
```
Octo.exe arg: create-release
Octo.exe arg: --project="""project nameI""".  <--  here
Octo.exe arg: --releaseNumber=""x.y.z"". <-- and here
Octo.exe arg: --channel="Default"
```

So in the last update, we updated [this lib](https://github.com/microsoft/azure-pipelines-task-lib), and it seems MS is doing some [weird things to do with quoting args](https://github.com/microsoft/azure-pipelines-task-lib/blob/9448530ccb0861863bbef53a52a41652f28ade66/node/toolrunner.ts#L395), so I need to figure out how to modify our code so we don't end up with this double-up of double-quotes.

**Comment by danielab** [comment]
That solves the issue! Thank you clarifying.
I see that the quotes are unnecessary (not sure why it was added back in time - maybe it was needed back then). However changes to an existing version should be backwards compatible, but i do understand that this was a dependency update issue which you be hard to anticipate.

However, I did find another pipeline in our azure devops that runs successfully with the double quotes. I cant see why that one is working. To me it does not really matter because removing the quotes works for the failing builds. But I have sent you the raw logs for that one as an email if you are interested for any fix (I do not need any response on that). 

Thank you for your help and quick response. Very appreciated.
